pkg/client/client.go

```diff
@@ -266,3 +268,38 @@ func buildCustomEndpointsMap(customEndpoints []configv1.AWSServiceEndpoint) map[
 
 	return customEndpointsMap
 }
+
+// sharedCredentialsFileFromSecret returns a location (path) to the shared credentials
+// file that was created using the provided secret
+func sharedCredentialsFileFromSecret(secret *corev1.Secret) (string, error) {
+	var data []byte
+	switch {
+	case len(secret.Data["credentials"]) > 0:
+		data = secret.Data["credentials"]
```

Comment:
Just for my understanding really, this credentials file, will it have a reference to the token that is being projected into the pod or does that come from somewhere else?

Reply:
It will have the path to the token file in it.  Will have to be in sync with the projected token location in https://github.com/openshift/machine-api-operator/pull/743.